tools/run_tests/dockerize/docker_run_tests.sh

```diff
@@ -42,7 +42,11 @@ export PATH=$PATH:/usr/bin/llvm-symbolizer
 chown $(whoami) $XDG_CACHE_HOME
 
 mkdir -p /var/local/git
-git clone --recursive /var/local/jenkins/grpc /var/local/git/grpc
+git clone /var/local/jenkins/grpc /var/local/git/grpc
+
+# clone gRPC submodules
+(cd /var/local/git/grpc/ && exec git submodule | awk '{ system("git submodule \
```

Comment:
What's here seems to be working but there seems to be some unnecessary complexity (the exec command), awk system, etc. I see they are there for a reason but it makes me wonder  - and others will probably wonder too, especially if they don't know what the idea behind this command is
how about
```
# clone each submodule using a local reference repo to avoid downloading entire submodules from github each time
git submodule foreach 'cd /var/local/jenkins/grpc && git submodule update --init --reference ../../jenkins/grpc/${name} ${name}'
```